### ISSUE TITLE
Missing level 1 heading

### ISSUE THREAD

**Issue by cyberbaloo** [suitenumerique/drive#427]
## Bug Report

**Problematic behavior**
The modal does not have a title and is therefore not properly rendered to assistive technologies

**Expected behavior/code**
`<div class="ReactModal__Content ReactModal__Content--after-open c__modal c__modal--small" tabindex="-1" role="dialog" aria-modal="true">` and `<h1 class="c__modal__title">Créer un dossier</h1>`

**Steps to Reproduce**
1. Opens Fichiers
2. Go to the menu “Mon Espace” 
3. And then the bug happens !

**Environment**
- Drive version:
- Platform:

**Possible Solution**
Modify the current title defined in the `<div class="c__modal__title"> Créer un dossier</div>` element to an `<h1>` element then add the `id="modal-title"` attribute

Add `aria-labelledby="modal-title"` to the `<div role="dialog" aria-modal="true”>` element.

**Additional context/Screenshots**

<img width="358" height="226" alt="Image" src="https://github.com/user-attachments/assets/aaa25d0b-f071-4702-b23e-5beaed8d0152" />
